### PR TITLE
Implemented for the request from the committee.

### PR DESCRIPTION
**Pull request by ageprocpp**
A member of the committee sent me two requests the day before yesterday and I implemented for it.
image

### CODE REVIEW

**Comment by caphosra** [comment]
Thank you for your contribution :)
I had already received the same requests from a member of the committee. But I ignored them because I had had a lot of tasks then :P

**Comment by ageprocpp** [comment]
I’ve just noticed that I removed ‘private’ in the declaration of ‘validLentItem’, but it was not necessary. I’m fixing it.

File: src/renderer/dataManager.ts [hunk 1]
```diff
@@ -84,9 +90,7 @@ export class DataManager<T extends YatsugiGroup | YatsugiTool> {
 
     async returnItem(toolIDs: string[]) {
         const isValid = await this.validReturnItem(toolIDs);
-        if (!isValid) {
-            throw "もう既に返却されている器材を返却しようとしています。";
-        }
+        if (!isValid) return;
```
Comment (caphosra) [inline src/renderer/dataManager.ts]: If an `async` function finishes unexpectedly, we must throw something.
Please revert this change and delete `await showErrorDialog(err);` from "returnTool.tsx".

```typescript
async returnButtonClicked() {
        try {
            await toolData.returnItem(this.state.tools.map(tool => tool.id));
            await showInfoDialog("返却完了しました。");
            this.props.history.push("/");
        }
        catch (err) {
            await showErrorDialog(err);
        }
    }
```

File: src/renderer/dataManager.ts [hunk 2]
```diff
@@ -116,7 +120,7 @@ export class DataManager<T extends YatsugiGroup | YatsugiTool> {
         await this.update();
     }
 
-    private async validLentItem(toolIDs: string[]): Promise<boolean> {
+    async validLentItem(toolIDs: string[]): Promise<boolean> {
```
Comment (caphosra) [inline src/renderer/dataManager.ts]: It seems that the `private` you wanted to remove is not this one.
And also it makes the latest build failed.

**Comment by ageprocpp** [comment]
We should throw an error when failed to return an item, but when the IDs include already returned item, shouldn't do so and continue processing as normal. I implemented `fixReturnItem` instead of `validReturnItem` in order not to have to throw an error.

**Comment by caphosra** [comment]
Nice! I'll merge this branch into master and publish this app with this changes.
